libs/coin-evm/src/api/node/ledger.common.ts

```diff
@@ -0,0 +1,352 @@
+  input: { address: string; contract: string }[],
+  params: { currency: CryptoCurrency },
+): Promise<BigNumber[]> => {
+  const { currency } = params;
+  const { node } = currency.ethereumLikeInfo || {};
+  if (!isLedgerNodeConfig(node)) {
+    throw new LedgerNodeUsedIncorrectly();
+  }
+
+  const balances = await fetchWithRetries<
+    Array<{
+      address: string;
+      contract: string;
+      balance: string;
+    }>
+  >({
+    method: "POST",
+    url: `${getEnv("EXPLORER")}/blockchain/v4/${node.explorerId}/erc20/balances`,
+    data: input,
+  });
+
+  return balances.map(({ balance }) => new BigNumber(balance));
+};
+
+/**
+ * Get the balance of an address for an ERC20 token
+ */
+export const getTokenBalance: RpcApi["getTokenBalance"] = async (
+  currency,
+  address,
+  contractAddress,
+) => {
+  const { node } = currency.ethereumLikeInfo || {};
+  if (!isLedgerNodeConfig(node)) {
+    throw new LedgerNodeUsedIncorrectly();
+  }
+
+  if (!tokenBalancesBatchersMap.has(currency)) {
+    const batcher = makeBatcher(getBatchTokenBalances, { currency });
+    tokenBalancesBatchersMap.set(currency, batcher);
+  }
+  const requestBatcher = tokenBalancesBatchersMap.get(currency)!;
+
+  return requestBatcher({ address, contract: contractAddress });
+};
+
+/**
+ * Get account nonce
+ */
+export const getTransactionCount: RpcApi["getTransactionCount"] = async (currency, address) => {
+  const { node } = currency.ethereumLikeInfo || {};
+  if (!isLedgerNodeConfig(node)) {
+    throw new LedgerNodeUsedIncorrectly();
+  }
+
+  const { nonce } = await fetchWithRetries<{
+    address: string;
+    nonce: number;
+  }>({
+    method: "GET",
+    url: `${getEnv("EXPLORER")}/blockchain/v4/${node.explorerId}/address/${address}/nonce`,
+  });
+
+  return nonce;
+};
+
+/**
+ * Get an estimated gas limit for a transaction
+ */
+export const getGasEstimation: RpcApi["getGasEstimation"] = async (account, transaction) => {
+  const { currency } = account;
+  const { node } = currency.ethereumLikeInfo || {};
+  if (!isLedgerNodeConfig(node)) {
+    throw new LedgerNodeUsedIncorrectly();
+  }
+
+  const { recipient: to, amount: value, data } = transaction;
+
+  try {
+    const { estimated_gas_limit: gasEtimation } = await fetchWithRetries<{
+      to: string;
+      estimated_gas_limit: string;
+    }>({
+      method: "POST",
+      url: `${getEnv("EXPLORER")}/blockchain/v4/${node.explorerId}/tx/estimate-gas-limit`,
+      data: {
+        from: account.freshAddress, // should be necessary for some estimations
+        to,
+        value: "0x" + (padHexString(value.toString(16)) || "00"),
+        data: data ? `0x${padHexString(data.toString("hex"))}` : "0x",
+      },
+    });
+
+    return new BigNumber(gasEtimation);
+  } catch (e) {
+    log("error", "EVM Family: Gas Estimation Error", e);
+    throw new GasEstimationError();
+  }
+};
+
+/**
+ * Get an estimation of fees on the network
+ */
+// Waiting for proxies
```

Comment:
This is a WIP ?

Reply:
That's the backends endpoints we're waiting for